pkg/alertmanager/statefulset.go

```diff
@@ -227,12 +227,7 @@ func makeStatefulSetSpec(a *monitoringv1.Alertmanager, config Config) (*appsv1.S
 		fmt.Sprintf("--config.file=%s", path.Join(alertmanagerConfigDir, alertmanagerConfigFile)),
 		fmt.Sprintf("--storage.path=%s", alertmanagerStorageDir),
 		fmt.Sprintf("--data.retention=%s", a.Spec.Retention),
-	}
-
-	if *a.Spec.Replicas == 1 && !a.Spec.ForceEnableClusterMode {
-		amArgs = append(amArgs, "--cluster.listen-address=")
-	} else {
-		amArgs = append(amArgs, "--cluster.listen-address=[$(POD_IP)]:9094")
```

Comment:
This logic ensures that alertmanager won't start clustering mode when running with only one replica. I don't see why this would need to be removed.
More: https://github.com/prometheus/alertmanager/blob/master/cmd/alertmanager/main.go#L231-L250

Reply:
Indeed, I did not properly read https://github.com/prometheus/alertmanager/blob/e3fb99cc2d24330fdb81aa1d466c19b3de328b5b/cmd/alertmanager/main.go#L231. Fixing.